lib/security/waf.ts

```diff
@@ -0,0 +1,120 @@
+import { Stack, StackProps } from 'aws-cdk-lib';
+import { ApplicationLoadBalancer } from 'aws-cdk-lib/aws-elasticloadbalancingv2';
+import { CfnWebACL, CfnWebACLAssociation, CfnWebACLAssociationProps } from 'aws-cdk-lib/aws-wafv2';
+import { Construct } from 'constructs';
+
+interface WafRule {
+    name: string;
+    rule: CfnWebACL.RuleProperty;
+}
+
+const awsManagedRules: WafRule[] = [
+  // AWS IP Reputation list includes known malicious actors/bots and is regularly updated
+  {
+    name: 'AWS-AWSManagedRulesAmazonIpReputationList',
+    rule: {
+      name: 'AWS-AWSManagedRulesAmazonIpReputationList',
+      priority: 0,
+      statement: {
+        managedRuleGroupStatement: {
+          vendorName: 'AWS',
+          name: 'AWSManagedRulesAmazonIpReputationList',
+        },
+      },
+      overrideAction: {
+        none: {},
+      },
+      visibilityConfig: {
+        sampledRequestsEnabled: true,
+        cloudWatchMetricsEnabled: true,
+        metricName: 'AWSManagedRulesAmazonIpReputationList',
+      },
+    },
+  },
+  // Blocks common SQL Injection
+  {
+    name: 'AWS-AWSManagedRulesSQLiRuleSet',
+    rule: {
+      name: 'AWS-AWSManagedRulesSQLiRuleSet',
+      priority: 1,
+      statement: {
+        managedRuleGroupStatement: {
+          vendorName: 'AWS',
+          name: 'AWSManagedRulesSQLiRuleSet',
+          excludedRules: [],
+        },
+      },
+      visibilityConfig: {
+        sampledRequestsEnabled: true,
+        cloudWatchMetricsEnabled: true,
+        metricName: 'AWS-AWSManagedRulesSQLiRuleSet',
+      },
+      overrideAction: {
+        none: {},
+      },
+    },
+  },
+  // Block request patterns associated with the exploitation of vulnerabilities specific to WordPress sites.
+  {
+    name: 'AWS-AWSManagedRulesWordPressRuleSet',
+    rule: {
+      name: 'AWS-AWSManagedRulesWordPressRuleSet',
+      priority: 2,
+      visibilityConfig: {
+        sampledRequestsEnabled: true,
+        cloudWatchMetricsEnabled: true,
+        metricName: 'AWS-AWSManagedRulesWordPressRuleSet',
+      },
+      overrideAction: {
+        none: {},
+      },
+      statement: {
+        managedRuleGroupStatement: {
+          vendorName: 'AWS',
+          name: 'AWSManagedRulesWordPressRuleSet',
+          excludedRules: [],
+        },
+      },
+    },
+  },
+];
+
+export class WAF extends CfnWebACL {
+  constructor(scope: Construct, id: string) {
+    super(scope, id, {
+      defaultAction: { allow: {} },
+      visibilityConfig: {
+        cloudWatchMetricsEnabled: true,
+        metricName: 'jenkins-WAF',
+        sampledRequestsEnabled: false,
```

Comment:
Why is this one disabled here for sampled request?

Reply:
Good catch. Enabled it now. Please check